test/stiefel.jl

```diff
@@ -296,4 +301,21 @@ include("utils.jl")
         q2 = retract(M, p, X, r2)
         @test is_manifold_point(M, q2)
     end
+
+    @testset "Canonical Metric" begin
+        M3 = MetricManifold(Stiefel(3, 2), CanonicalMetric())
+        p = [1.0 0.0; 0.0 1.0; 0.0 0.0]
+        X = [0.0 0.0; 0.0 0.0; 1.0 1.0]
+        q = exp(M3, p, X)
+        Y = [0.0 0.0; 0.0 0.0; -1.0 1.0]
+        r = exp(M3, p, Y)
+        @test isapprox(M3, p, log(M3, p, q), X)
+        @test isapprox(M3, p, log(M3, p, r), Y)
+        @test inner(M3, p, X, Y) == 0
```

Comment:
Could you also add a test for `inner` for non-orthogonal vectors?

Reply:
It's sometimes hard to come up with numbers,

I also need a test that actually needs the iterative process, currently all `log`s are already precise enough after the procrustes prestep; I just can not find a good example where that Is needed. If you have a good idea, feel free to propose a test (see currently the 4 uncovered lines).

Reply:
I think you could just try something like `inner(M3, p, X, 2*X+3*Y)`? I have no idea how to test that case of `log` though other than randomly trying different points until the right code path is hit.

Reply:
Ok, I will add that, but what should I compare that to?

As I wrote above, I even already talked to Ralf about that, I will try to find one.

Reply:
Well, it should be equal to `2*norm(M3, p, X)^2` I think.

Reply:
Ah, I alredy forgot about the orthogonality of X and Y, sorry; yes, you're right. I will add that.

Reply:
For the Euclidean distance, I changed transparency and added a test, that it really does not work.
We can change that of course once we get a log for Euclidean metric.